### ISSUE TITLE
IsInitOnly boolean property for PropertyDefinition

### ISSUE THREAD

**Issue by dbeylkhanov** [dotnet/sdk#29616]
### Is your feature request related to a problem? Please describe.

I am trying to check that my property has `init`-only setter 

### Describe the solution you'd like
it would be nice to have IsInitOnly method similar to FieldDefinition class from Mono.Cecil  

### Additional context

Below the IsInitOnly property in FieldDefinition
```
public bool IsInitOnly
    {
      get => this.attributes.GetAttributes((ushort) 32);
      set => this.attributes = this.attributes.SetAttributes((ushort) 32, value);
    }
```

**Comment by dbeylkhanov** [comment]
> https://github.com/dotnet/runtime/issues/43088#issuecomment-704304906

yeah, I saw this comment before the opening this issue. I think that the suggested solution is a cratch. I don't want use that in my tests

**Comment by nagilson** [comment]
May you elaborate as to why it's a crutch and why you want that property in particular? Probably we will move this elsewhere, maybe the Runtime, doesn't seem like it's part of the SDK.
